apps/researcher/src/lib/date-formatter.ts

```diff
@@ -0,0 +1,34 @@
+'use server';
+
+import {getLocale, getTranslations} from 'next-intl/server';
+
+export async function formatDate(date?: Date) {
+  const t = await getTranslations('FormatDate');
+  const locale = await getLocale();
+  const formatter = new Intl.DateTimeFormat(locale, {
+    year: 'numeric',
+    month: 'short',
+    day: 'numeric',
+  });
```

Comment:
Out of curiosity: are there performance considerations for getting translations, locales and creating new formatter instances every time `formatDate` is called? Is it necessary to pre-fetch or cache these or doesn't it have a noticeable perf impact?

Reply:
Yes, there can be performance considerations when fetching translations and locales and creating new formatter instances every time `formatDate` is called. Creating a new `Intl.DateTimeFormat` instance is also not free, although it's likely less expensive than fetching translations or locales.

Taking this into consideration, I'm going back to the old design. I will call the function `getFormatDate`.

Reply:
Yes, that makes sense!

A final consideration: if we, at some point, want to call `getFormatDate()` in different places/components, every call will load the translations and the locale and create a datetime formatter. Perhaps we should then refactor and put these objects in some global place (e.g. a [registry](https://martinfowler.com/eaaCatalog/registry.html)).